File: common/src/main/java/com/pluxity/user/repository/RoleRepository.java
package com.pluxity.user.repository;

import com.pluxity.user.entity.Role;
import jakarta.annotation.Nonnull;
import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;

public interface RoleRepository extends JpaRepository<Role, Long> {
    Optional<Role> findByName(String name);

    @EntityGraph(
            attributePaths = {
                "userRoles.user",
                "userRoles.role",
                "rolePermissions.permissionGroup.permissions"
            })
    @Override
    @Nonnull
    Optional<Role> findById(@Nonnull Long id);

    @EntityGraph(
            attributePaths = {
                "userRoles.user",
                "userRoles.role",
                "rolePermissions.permissionGroup.permissions"
            })
    @Override
    @Nonnull
    List<Role> findAll();
}

File: common/src/main/kotlin/com/pluxity/user/repository/RoleRepository.kt
package com.pluxity.user.repository

import com.pluxity.user.entity.Role
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository

interface RoleRepository : JpaRepository<Role, Long> {
    @EntityGraph(
        attributePaths = [
            "userRoles.user", "userRoles.role", "rolePermissions.permissionGroup.permissions", "roleGlobalPolicies",
        ],
    )
    fun findWithInfoById(id: Long): Role?

    @EntityGraph(
        attributePaths = [
            "userRoles.user", "userRoles.role", "rolePermissions.permissionGroup.permissions", "roleGlobalPolicies",
        ],
    )
    fun findByAuthIsNotOrderByCreatedAtDesc(auth: String): List<Role>
}
